db/database.go

```diff
@@ -418,8 +423,20 @@ func NewDatabaseContext(ctx context.Context, dbName string, bucket base.Bucket,
 	metaKeys := base.NewMetadataKeys(options.MetadataID)
 	dbContext.MetadataKeys = metaKeys
 
+	// Initialize JavaScript VMPool:
+	jsEngineName := DefaultJavaScriptEngine
+	if options.JavaScriptEngine != nil {
+		jsEngineName = *options.JavaScriptEngine
+	}
+	if engine := js.EngineNamed(jsEngineName); engine != nil {
+		dbContext.JS.Init(engine, MaxJavaScriptVMs)
+	} else {
+		return nil, fmt.Errorf("%q is not an available JavaScript engine", jsEngineName)
+	}
+
 	cleanupFunctions = append(cleanupFunctions, func() {
```

Comment:
you need to move the 

		base.SyncGatewayStats.ClearDBStats(dbName)

to above initializing js engine because we have the unlikely potential to require an error.

Reply:
yup.